app/build.gradle

```diff
@@ -268,3 +257,35 @@ task printGeckoviewVersions {
 // Normally this should use the same version as the dependency. But since we are using snapshots we can't reference a
 // git tag. So we are just using "master" and hoping for the best.
 apply from: 'https://github.com/mozilla-mobile/android-components/raw/master/components/service/glean/scripts/sdk_generator.gradle'
+
+// Internal, but stable and convenient.
+import org.gradle.api.internal.artifacts.DefaultModuleIdentifier
+
+// This is the equivalent of what the `org.mozilla.appservices` Gradle plugin
+// achieves, minus asserting that the versions of all the megazord components
+// agree.  This could be added if required.
+afterEvaluate {
+    def megazord = DefaultModuleIdentifier.newId('org.mozilla.appservices', 'reference-browser-megazord')
+    def megazordComponents = [
+            DefaultModuleIdentifier.newId('org.mozilla.appservices', 'fxaclient'),
+            DefaultModuleIdentifier.newId('org.mozilla.appservices', 'places'),
+            DefaultModuleIdentifier.newId('org.mozilla.appservices', 'push'),
+            DefaultModuleIdentifier.newId('org.mozilla.appservices', 'rustlog'),
+            DefaultModuleIdentifier.newId('org.mozilla.appservices', 'viaduct')
+    ]
+
+    configurations.each { configuration ->
+        configuration.resolutionStrategy.dependencySubstitution.all { dependency ->
+            if (dependency.requested instanceof ModuleComponentSelector) {
+                def moduleIdentifier = ((ModuleComponentSelector) dependency.requested).moduleIdentifier
+                if (megazordComponents.contains(moduleIdentifier)) {
+                    def megazordVersion = Deps.mozilla_appservices
+                    def substitution = "${megazord.group}:${megazord.name}:${megazordVersion}"
```

Comment:
I see why you were asking about the Megazord version.  You could use `dependency.requested.version` here, and potentially have clashing versions be resolved by Gradle.  Then you wouldn't need to specify in the consumer.  That's pretty safe, tbh.